tests/test_examples.py

```diff
@@ -1200,6 +1204,210 @@ def test_example_get_current(self):
+        # and we want this to provide half the field the surface current's
+        # G is providing, in the same direction
+        external_field=ToroidalMagneticField(B0=-mu_0 * (G / 2) / 2 / np.pi, R0=1),
+        vacuum=True,
+    )
+    phi_mn_opt = surface_current_field.Phi_mn
+    np.testing.assert_allclose(G / 2, surface_current_field.G, atol=1e-8)
+    np.testing.assert_allclose(phi_mn_opt, 0, atol=1e-10)
+    np.testing.assert_allclose(
+        surface_current_field.compute("Phi", grid=grid)["Phi"],
+        correct_phi / 2,
+        atol=1e-9,
+    )
+    np.testing.assert_allclose(data["chi^2_B"], 0, atol=1e-11)
+    B_from_surf = surface_current_field.compute_magnetic_field(
+        coords, source_grid=LinearGrid(M=200, N=200, NFP=surf_winding.NFP)
+    )
+    np.testing.assert_allclose(B, B_from_surf * 2, atol=1e-4)
+
+
+@pytest.mark.regression
+@pytest.mark.solve
+@pytest.mark.slow
+def test_regcoil_modular_check_B(regcoil_modular_coils):
+    """Test precise QA modular (helicity=0) regcoil solution."""
+    (
+        data,
+        initial_surface_current_field,
+        eq,
+    ) = regcoil_modular_coils
+    chi_B = data["chi^2_B"]
+    surface_current_field = initial_surface_current_field.copy()
+
+    np.testing.assert_array_less(chi_B, 1e-5)
+    coords = eq.compute(["R", "phi", "Z", "B"])
+    B = coords["B"]
+    coords = np.vstack([coords["R"], coords["phi"], coords["Z"]]).T
+    B_from_surf = surface_current_field.compute_magnetic_field(
+        coords,
+        source_grid=LinearGrid(M=80, N=80, NFP=surface_current_field.NFP),
+        basis="rpz",
+    )
+    np.testing.assert_allclose(B, B_from_surf, atol=3e-3)
+
+
+@pytest.mark.regression
+@pytest.mark.solve
+@pytest.mark.slow
+def test_regcoil_helical_coils_check_objective_method(
+    regcoil_helical_coils_scan,
+):
+    """Test precise QA helical coil regcoil solution."""
+    M_egrid = 30
+    N_egrid = 30
+    M_sgrid = 50
+    N_sgrid = 50
+
+    (
+        _,
+        initial_surface_current_field,
+        eq,
+    ) = regcoil_helical_coils_scan
+    surface_current_field = initial_surface_current_field.copy()
+
+    # reset the Phi_mn
+    surface_current_field.Phi_mn = surface_current_field.Phi_mn.at[:].set(0.0)
+    constraints = (  # now fix all but Phi_mn
+        FixParameters(
+            surface_current_field,
+            params={"I": True, "G": True, "R_lmn": True, "Z_lmn": True},
+        ),
+    )
+
+    eval_grid = LinearGrid(M=M_egrid, N=N_egrid, NFP=eq.NFP, sym=True)
+    sgrid = LinearGrid(
+        M=M_sgrid,
+        N=N_sgrid,
+        NFP=eq.NFP,
+    )
+
+    obj = QuadraticFlux(
+        field=surface_current_field,
+        eq=eq,
+        eval_grid=eval_grid,
+        field_grid=sgrid,
+        vacuum=True,
+    )
+
+    objective = ObjectiveFunction(
+        (
+            obj,
+            SurfaceCurrentRegularization(
+                surface_current_field=surface_current_field,
+                weight=np.sqrt(1e-18),
```

Comment:
doesn't the original surface have lambda=0?

Reply:
changed to match same one being compared to